utils/utils.go

```diff
@@ -0,0 +1,43 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func GetOwnerRepoPull(ref, repo string) (string, string, int) {
+	// get pull request files
+	githubrefS := strings.Split(ref, "/")
+	prnum := githubrefS[2]
+	pull, _ := strconv.Atoi(prnum)
+
+	repoS := strings.Split(repo, "/")
+	owner := repoS[0]
+	repoName := repoS[1]
+
+	return owner, repoName, pull
+}
+
+func ValidateModuleSource(source string, approvedModules map[string]bool) (bool, error) {
+	for k, v := range approvedModules {
+		if strings.Contains(source, k) {
+			if v {
+				return true, nil
+			}
+		}
+	}
+	return false, fmt.Errorf("module not approved")
+}
+
+// get source line from source file
+func GetSourceLine(source string) string {
+	sourceS := strings.Split(source, "\n")
+	for _, line := range sourceS {
+		if strings.Contains(line, "source") {
+			sourceLine := strings.Split(line, "=")
+			return sourceLine[1]
```

Comment:
this will return the first match for source line and not loop through others is that what we want?

Reply:
source will only be one file, you would feed the sources files in for loop in your main file of your script
```
for _, file := range files {
...
 utils.GetSourceLine(file)
}
```